### PR TITLE
feat(argo-cd): Add revisionHistoryLimit

### PR DESCRIPTION
**Pull request by pdrastil**
Closes: https://github.com/argoproj/argo-helm/issues/1578

---

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -2839,10 +2842,6 @@ notifications:
       ## You have to set secret.notifiers.slack.signingSecret
       enabled: false
 
-      # -- The deployment strategy to use to replace existing pods with new ones
```
Comment (pdrastil) [inline charts/argo-cd/values.yaml:2842]: Removed as SlackBot must be always recreated first.

File: charts/argo-cd/values.yaml [hunk 2]
```diff
@@ -46,6 +46,9 @@ global:
   additionalLabels: {}
     # app: argo-cd
 
+  # -- Number of old deployment ReplicaSets to retain. The rest will be garbage collected.
+  revisionHistoryLimit: 3
```
Comment (mkilchhofer) [inline charts/argo-cd/values.yaml:50]: Lets set it to 5 again so we can also update the STS of the controller.

Comment (pdrastil) [reply charts/argo-cd/values.yaml:50]: STS does not support this - it's there from historic reasons because it was Deployment 1 year ago.

Comment (pdrastil) [reply charts/argo-cd/values.yaml:50]: Also if you try to remove history limit from STS you get following:

```sh
The StatefulSet "argocd-application-controller" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden
```

Only way to upgrade is to run these and I wanted to keep this for 6.x release when we remove old things.

```sh
kubectl delete argocd-application-controller
helm upgrade argocd
```

Comment (mkilchhofer) [reply charts/argo-cd/values.yaml:50]: Ah you are right 👍

**Review by mkilchhofer — APPROVED** [review]
Lgtm

File: charts/argo-cd/values.yaml (hunk 2 above)
Comment (mbevc1) [reply charts/argo-cd/values.yaml:50]: Any reason why we're not matching 5 here as a default?

Comment (pdrastil) [reply charts/argo-cd/values.yaml:50]: I've reduced this to 3 to make UI less cluttered when Argo CD is watching itself. Users can adjust amount of replicas so if you want more you can adjust it.

Comment (mbevc1) [reply charts/argo-cd/values.yaml:50]: I'm happy with this, but I'd try to align to previous or upstream defaults if we can.

**Pull request merged**